File: chat-uikit/src/main/java/io/agora/chat/uikit/interfaces/IRecyclerView.java
package io.agora.chat.uikit.interfaces;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;


public interface IRecyclerView {

    /**
     * Add header adapter
     * @param adapter
     */
    void addHeaderAdapter(RecyclerView.Adapter adapter);

    /**
     * Add footer adapter
     * @param adapter
     */
    void addFooterAdapter(RecyclerView.Adapter adapter);

    void removeAdapter(RecyclerView.Adapter adapter);

    void addRVItemDecoration(@NonNull RecyclerView.ItemDecoration decor);

    void removeRVItemDecoration(@NonNull RecyclerView.ItemDecoration decor);
    
    default void setOnItemClickListener(OnItemClickListener listener) {}
    
    default void setOnItemLongClickListener(OnItemLongClickListener listener) {}
}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/common/interfaces/IRecyclerView.kt
package io.agora.uikit.common.interfaces

import androidx.recyclerview.widget.RecyclerView
import io.agora.uikit.interfaces.OnItemClickListener
import io.agora.uikit.interfaces.OnItemLongClickListener
import io.agora.uikit.interfaces.OnRecyclerViewTouchListener

/**
 * Define the common methods of RecyclerView.
 */
interface IRecyclerView {
    /**
     * Add header adapter
     * @param adapter
     */
    fun addHeaderAdapter(adapter: RecyclerView.Adapter<*>?)

    /**
     * Add footer adapter
     * @param adapter
     */
    fun addFooterAdapter(adapter: RecyclerView.Adapter<*>?)

    /**
     * Remove adapter.
     * @param adapter
     */
    fun removeAdapter(adapter: RecyclerView.Adapter<*>?)

    /**
     * Add item decoration.
     * @param decor
     */
    fun addItemDecoration(decor: RecyclerView.ItemDecoration)

    /**
     * Remove item decoration.
     * @param decor
     */
    fun removeItemDecoration(decor: RecyclerView.ItemDecoration)

    /**
     * Set item click listener.
     */
    fun setOnItemClickListener(listener: OnItemClickListener?) {}

    /**
     * Set item long click listener.
     */
    fun setOnItemLongClickListener(listener: OnItemLongClickListener?) {}

    /**
     * Set recyclerView touch listener.
     */
    fun setOnRecyclerViewTouchListener(listener: OnRecyclerViewTouchListener?) {}

    /**
     * Refresh message list.
     */
    fun notifyDataSetChanged() {}

}